File: app/src/main/java/solutions/alterego/android/unisannio/ingegneria/IngegneriaCercapersoneParser.java
package solutions.alterego.android.unisannio.ingegneria;

import android.util.Log;

import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import rx.Observable;
import solutions.alterego.android.unisannio.cercapersone.Person;
import solutions.alterego.android.unisannio.interfaces.IParser;

public class IngegneriaCercapersoneParser implements IParser<Person> {

    @Inject
    public IngegneriaCercapersoneParser() {
    }

    @Override
    public ArrayList<Person> parse(Document document) {

        ArrayList<Person> persons = new ArrayList<>();

        Elements elements = document.getElementsByTag("item");

        for (Element element:elements){

                String nome = element.getElementsByTag("title").text();
                String ruolo = element.getElementsByTag("position").text();
                String email = element.getElementsByTag("email").text();
                String ufficio = element.getElementsByTag("address").text();
                String telefono = element.getElementsByTag("telephone").text();
                String webPage = element.getElementsByTag("personalwebpage").text();
                String tutoring = element.getElementsByTag("tutoring").text();

            if(nome != null) {
                Person person = new Person(nome, ruolo, email, telefono, ufficio, webPage, tutoring);
                persons.add(person);
            }
        }
                return persons;
    }
}

File: app/src/main/java/solutions/alterego/android/unisannio/ingegneria/IngegneriaCercapersoneParser.kt
package solutions.alterego.android.unisannio.ingegneria

import org.jsoup.nodes.Document
import org.jsoup.nodes.Element
import org.jsoup.select.Elements

import java.util.ArrayList

import javax.inject.Inject

import solutions.alterego.android.unisannio.cercapersone.Person
import solutions.alterego.android.unisannio.interfaces.Parser

class IngegneriaCercapersoneParser @Inject
constructor() : Parser<Person> {

    override fun parse(document: Document): ArrayList<Person> {

        val persons = ArrayList<Person>()

        val elements = document.getElementsByTag("item")

        for (element in elements) {

            val nome = element.getElementsByTag("title").text()
            val ruolo = element.getElementsByTag("position").text()
            val email = element.getElementsByTag("email").text()
            val ufficio = element.getElementsByTag("address").text()
            val telefono = element.getElementsByTag("telephone").text()
            val webPage = element.getElementsByTag("personalwebpage").text()
            val tutoring = element.getElementsByTag("tutoring").text()

            if (nome != null) {
                val person = Person(nome, ruolo, email, telefono, ufficio, webPage, tutoring)
                persons.add(person)
            }
        }
        return persons
    }
}
